charts/consul/values.yaml

```
@@ -117,25 +117,56 @@ global:
   # created by this chart. See https://kubernetes.io/docs/concepts/policy/pod-security-policy/.
   enablePodSecurityPolicies: false
 
-  # Configures Consul's gossip encryption key, set as a Kubernetes secret
+  # secretsBackend is used to configure Vault as the secrets backend for the Consul on Kubernetes installation.
+  # The Vault cluster needs to have the Kubernetes Auth Method, 
+  # KV2 and PKI secrets engines enabled and have necessary secrets, 
+  # policies and roles created prior to installing Consul.
+  # The Vault cluster should not have Consul as its storage backend.
+  # Note: When using Vault KV2 secrets engines the "data" field is implicitly required for Vault API calls,
+  # secretName should be in the form of  "vault-kv2-mount-path/data/secret-name".
+  # secretKey should be in the form of "key".
+  secretsBackend:
+    vault:
+      # Enabling the Vault secrets backend will replace Kubernetes secrets with referenced Vault secrets.
+      enabled: false
+
+      # The consulServerRole must be created and have read capabilties via a Policy for the following secrets:
+      # - gossip encryption key defined by `global.gossipEncryption.secretName`.
```

Adding a bit more detail. If you like it, it'd be nice to do the same for the client role.
```suggestion
      # The Vault role for the Consul server.
      # The role must be connected to the Consul server's service account and
      # have have a policy with read capabilities for the following secrets:
      # - gossip encryption key defined by `global.gossipEncryption.secretName`.
      # To discover the service account name of the Consul server, run
      #     ```
      #     helm template -s templates/server-serviceaccount.yaml <release-name> charts/consul
      #     ```
      # and check the name of `metadata.name`.
```